src/extensions/agent_modules/shared/upd_utils/src/adu_upd_utils.c

```diff
@@ -127,4 +128,18 @@ void AduUpdUtils_TransitionState(ADU_UPD_STATE newState, ADUC_Update_Request_Ope
         newState,
         AduUpdState_str(newState));
     updateData->updState = newState;
+
+    if (newState == ADU_UPD_STATE_IDLEWAIT)
+    {
+        srand(time(NULL));
+        int jitter_seconds = (rand() % 121) + 60; // between 1 and 3 minutes of jitter
```

Comment:
Should be a #define centralized in a seperate .h file for all these random numbers.